Rodi add stop action

tchx84 opened the pull request.

tchx84 commented:
@walterbender I noticed something, each time the plugin is loaded a new listener is added, and those listener stack up, making unnecessary calls.

To reproduce:
1. open the developer console, clean the log and look.
2. load the plugin, and press stop.
3. now, remove the plugin and press stop.
4. re-load the plugin, and press stop.

Repeat this, and you will see that the listeners stack up, which is bad news.

I assume the global section is executed each time the plugin is added, but there is no section for when the plugin is removed, I think we need something like that to "undo" global changes. e.g, I would use removeEventListener there to solve this.

The other walkaround could be to detect if the listener is already there, but I guess that won't work because the function references will be different each time.

walterbender commented:
Sounds like a good GCI project :)

On Thu, Oct 22, 2015 at 2:47 PM, Martin Abente Lahaye <
notifications@github.com> wrote:

> @walterbender <https://github.com/walterbender> I noticed something, each
> time the plugin is loaded a new listener is added, and those listener stack
> up, making unnecessary calls.
>
> To reproduce:
> 1. open the developer console, clean the log and look.
> 2. load the plugin, and press stop.
> 3. now, remove the plugin and press stop.
> 4. re-load the plugin, and press stop.
>
> Repeat this, and you will see that the listeners stack up, which is bad
> news.
>
> I assume the global section is executed each time the plugin is added, but
> there is no section for when the plugin is removed, I think we need
> something like that to "undo" global changes. e.g, I would use
> removeEventListener there to solve this.
>
> The other walkaround could be to detect if the listener is already there,
> but I guess that won't work because the function references will be
> different each time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/walterbender/turtleblocksjs/pull/241#issuecomment-150319996>
> .
>



-- 
Walter Bender
Sugar Labs
http://www.sugarlabs.org

tchx84 commented:
I cleaned up the code, fixed a couple more bugs, so I have a new PR [1] which includes this PR. So I am closing this.

[1] https://github.com/walterbender/turtleblocksjs/pull/242
